check/src/kindcheck.rs

```diff
@@ -171,6 +171,7 @@ impl<'a> KindCheck<'a> {
             }
             Type::Variable(_) => Ok((self.subs.new_var(), typ.clone())),
             Type::Builtin(builtin_typ) => Ok((self.builtin_kind(builtin_typ), typ.clone())),
+            Type::Forall(_, _) => unimplemented!(),
```

Comment:
No, it should be fine to just instantiate the type with `vars` as the variables themselves. As long as their kinds are correct the resulting kind will be correct.